yt_dlp/extractor/videocampus_sachsen.py

```diff
@@ -157,30 +197,33 @@ def _real_extract(self, url):
             'host', 'id', 'tmp_id', 'display_id', 'embed_id')
         webpage = self._download_webpage(url, video_id or tmp_id, fatal=False) or ''
 
-        if not video_id:
-            video_id = embed_id or self._html_search_regex(
-                rf'src="https?://{host}/media/embed.*(?:\?|&)key=([0-9a-f]+)&?',
-                webpage, 'video_id')
+        formats, subtitles = [], {}
+        title = description = thumbnail = None
+
+        metadata = self._search_json(r'var\s+options\s*=\s*', webpage, 'player options', video_id or tmp_id or embed_id, default=None, fatal=False)
+
+        if metadata:
+            for source in metadata.get('sources', []):
+                if source.get('type') == 'application/x-mpegURL' and source.get('src'):
+                    _formats, _subtitles = self._extract_m3u8_formats_and_subtitles(source.get('src'), video_id or tmp_id or embed_id, fatal=False)
+                    formats.extend(_formats)
+                    subtitles.update(_subtitles)
+                elif source.get('src'):
+                    formats.append({'url': source.get('src')})
```

Comment:
- use `traverse_obj` and `mimetype2ext` (import from utils)
- no point to the `_`-prefixed temp vars, we can just overwrite `formats` and `subtitles`
```suggestion
            for source in traverse_obj(metadata, ('sources', lambda _, v: url_or_none(v['src']))):
                if mimetype2ext(source.get('type')) == 'm3u8':
                    formats, subtitles = self._extract_m3u8_formats_and_subtitles(source['src'], display_id, 'mp4')
                else:
                    formats.append({'url': source['src']})
```